edx_lint/pylint/yaml_load_check.py

```diff
@@ -0,0 +1,49 @@
+"""
+Checks for unsafe ``yaml.load()`` calls.
+The PyYAML library has not yet released a version
+that addresses the vulnerability (listed below) in the commonly
+called ``load()`` function.
+
+Vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2017-18342
+PyYAML release plan: https://github.com/yaml/pyyaml/issues/193
+
+When a new release is published, we may be able to remove this check.
+"""
+
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+
+from .common import BASE_ID, check_visitors
+
+
+def register_checkers(linter):
+    """Register checkers."""
+    linter.register_checker(YamlLoadChecker(linter))
+
+
+@check_visitors
+class YamlLoadChecker(BaseChecker):
+    """
+    Checks for unsafe ``yaml.load()`` calls.
+    """
+
+    __implements__ = (IAstroidChecker,)
+
+    name = 'yaml-load-checker'
+
+    MESSAGE_ID = 'unsafe-yaml-load'
+
+    UNSAFE_CALL = 'yaml.load'
```

Comment:
Can we also check for the lesser-used "load_all"?  BTW, there's also "safe_dump" and "safe_dump_all", though I'm less clear on what the dangers are with dump?
```
>>> [n for n in dir(yaml) if "safe" in n]
['safe_dump', 'safe_dump_all', 'safe_load', 'safe_load_all']
```

Reply:
I think the "safe_*" functions are meant to be safe to use.  The goal of this checker is to warn against `load()` and have the developer use `safe_load()` instead.  At least, that was my reading of the PyYAML release plan and the CVE.

Reply:
We can check for `load_all`, though - good call on that.